views/js/component/diagnostic/diagnosticLoader.js

```diff
@@ -69,7 +70,20 @@ define([
                             factoryConfig = componentConfig[factoryName];
                             factoryConfig.controller = componentConfig.controller;
 
-                            factory(self.getElement(), factoryConfig);
+                            var componentInstance = factory(self.getElement(), factoryConfig);
+
+                            componentInstance
+                                .on('render', function() {
+                                    if (factoryConfig.autoStart) {
+                                        this.run();
+                                    }
+                                })
+                                .on('start', function() {
+                                    loadingBar.start();
```

Comment:
`loadingBar` should not be managed by a component, since this is related to the page. This coupling will lead to issues, so please leave this in the constructor.

Instead, you should rather forward the events so that the controller will work properly:
```javascript
factory(self.getElement(), factoryConfig)
    .on('render', function() {
        if (factoryConfig.autoStart) {
            this.run();
        }
    })
    .on('start', function() {
        self.trigger('start', factoryName);
    })
    .on('end', function() {
        self.trigger('stop', factoryName);
    });
```

Reply:
Same here :(